File: backend/app/common/enums.py
from enum import StrEnum


class UserRole(StrEnum):
    ADMIN = "admin"
    CUSTOMER = "customer"


class PurposeOTP(StrEnum):
    LOGIN = "login"
    REGISTER = "register"
    RESET = "reset"


class InventoryStatus(StrEnum):
    IN_STOCK = "in_stock"
    LOW_STOCK = "low_stock"
    OUT_OF_STOCK = "out_of_stock"


class ProductStatus(StrEnum):
    ACTIVE = "active"
    INACTIVE = "inactive"
    DRAFT = "draft"
    ARCHIVED = "archived"


class ProductSortEnum(StrEnum):
    NEWEST = "newest"
    PRICE_ASC = "price_asc"
    PRICE_DESC = "price_desc"
    DISCOUNT_DESC = "discount_desc"

File: Axum/src/models/enums.rs
// src/models/enums.rs 
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, sqlx::Type)]
#[sqlx(type_name = "inventorystatus", rename_all = "UPPERCASE")]
pub enum InventoryStatus {
    InStock,
    LowStock,
    OutOfStock,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, sqlx::Type)]
#[sqlx(type_name = "productstatus", rename_all = "UPPERCASE")]
pub enum ProductStatus {
    Active,
    Inactive,
    Draft,
    Archived,
}

// #[derive(Debug, Clone, PartialEq, Eq, Type, Serialize, Deserialize)]
// #[sqlx(type_name = "cartstatus", rename_all = "lowercase")]
// pub enum CartStatus {
//     Active,
//     Abandoned,
//     Converted,
// }

// #[derive(Debug, Clone, PartialEq, Eq, Type, Serialize, Deserialize)]
// #[sqlx(type_name = "userrole", rename_all = "lowercase")]
// pub enum UserRole {
//     Admin,
//     Customer,
// }

// #[derive(Debug, Clone, PartialEq, Eq, Type, Serialize, Deserialize)]
// #[sqlx(type_name = "purposeotp", rename_all = "lowercase")]
// pub enum PurposeOtp {
//     Login,
//     Register,
//     Reset,
// }
